tasks/travis-pr-title-lint.js

```diff
@@ -135,20 +168,34 @@ module.exports = function(grunt) {
           objectEntries.shim();
         }
 
-        countGitMergeCommits()
-          .then(function(commitsCount) {
-            if (commitsCount === 1) {
+        getGitBranchCommits()
+          .then(function(commits) {
+            if (commits.length === 1) {
               grunt.log.writeln(
                 'There is only one commit in this pull request, ' +
                 'we are going to check the single commit message...'
               );
-              return getGitLastCommitMessage().then(lintMessage);
+              return getGitCommitMessage(commits[0]).then(lintMessage);
             } else {
               grunt.log.writeln(
                 'There is more than one commit in this pull request, ' +
                 'we are going to check the pull request title...'
               );
-              return getPullRequestTitle().then(lintMessage);
+
+              return getPullRequestTitle()
+                .then(function(pullRequestTitle) {
+                  if (!pullRequestTitle) {
```

Comment:
@kumar303 I added here the "retry to get the pull request title from the github API if it is empty" to reduce the intermittent failures related to this weird behavior of the github API requests when this task is running on travis.